internal/provider/resource_hvn_route.go

```diff
@@ -1,14 +1,293 @@
+				Computed:    true,
+			},
+			// TODO add updated_at after public API returns updated_at for HVN route
+		},
+	}
+}
+
+func resourceHvnRouteCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	client := meta.(*clients.Client)
+
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+
+	destination := d.Get("destination_cidr").(string)
+	hvnRouteID := d.Get("hvn_route_id").(string)
+
+	hvn := d.Get("hvn").(string)
+	var hvnLink *sharedmodels.HashicorpCloudLocationLink
+	hvnLink, err := buildLinkFromURL(hvn, HvnResourceType, loc.OrganizationID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	target := d.Get("target_link").(string)
+	var targetLink *sharedmodels.HashicorpCloudLocationLink
+	// If target type is neither a peering nor a TGW attachment, we will return an error.
+	// Currently create HVN route will only support these two resource types.
+	targetLink, err = buildLinkFromURL(target, PeeringResourceType, loc.OrganizationID)
+	if err != nil {
+		targetLink, err = buildLinkFromURL(target, TgwAttachmentResourceType, loc.OrganizationID)
+		if err != nil {
+			return diag.Errorf("unable to parse target_link for HVN (%s) route with destination CIDR of %s: %v; target must be of either %s or %s", hvnLink.ID, destination, err, PeeringResourceType, TgwAttachmentResourceType)
+		}
+	}
+
+	// Check for an existing HVN.
+	retrievedHvn, err := clients.GetHvnByID(ctx, client, loc, hvnLink.ID)
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			return diag.Errorf("unable to find the HVN (%s) for the HVN route", hvnLink.ID)
+		}
+
+		return diag.Errorf("unable to check for presence of an existing HVN (%s): %v", hvnLink.ID, err)
+	}
+
+	if retrievedHvn.State != networkmodels.HashicorpCloudNetwork20200907NetworkStateSTABLE {
+		return diag.Errorf("unable to create HVN route for HVN (%s), HVN state is not STABLE", retrievedHvn.ID)
+	}
+
+	log.Printf("[INFO] HVN (%s) found, proceeding with HVN route create", hvnLink.ID)
+
+	// Check if HVN route already exists.
+	_, err = clients.ListHVNRoutes(ctx, client, hvnLink.ID, destination, "", "", loc)
+	if err != nil {
+		if !clients.IsResponseCodeNotFound(err) {
+			return diag.Errorf("unable to check for presence of an existing route for HVN (%s) with the destination CIDR of %s: %v", hvnLink.ID, destination, err)
+		}
+
+		return diag.Errorf("an HVN route with destination=%s, hvn_id=%s and project_id=%s already exists - to be managed via Terraform this resource needs to be imported into the state. Please see the resource documentation for hcp_hvn_route for more information", destination, hvnLink.ID, loc.ProjectID)
+	} else {
+		log.Printf("[INFO] HVN route with destination CIDR of %s for HVN (%s) not found, proceeding with HVN route create", destination, hvnLink.ID)
+	}
+
+	// Check if the target exists.
+	switch targetLink.Type {
```

Comment:
same, let's please delete it, it's not a concern of TF provider to do such a business-logic specific validation

Reply:
same, discussed with Brenna on zoom and agreed that this is unnecessary here to duplicate backend logic